src/arch/src/aarch64/regs.rs

```diff
@@ -176,6 +207,237 @@ pub fn read_mpidr(vcpu: &VcpuFd) -> Result<u64> {
+        .map_err(Error::SetCoreRegister)?;
+
+    let off = offset__of!(kvm_regs, sp_el1);
+    vcpu.set_one_reg(arm64_core_reg_id!(KVM_REG_SIZE_U64, off), state.sp_el1)
+        .map_err(Error::SetCoreRegister)?;
+
+    let off = offset__of!(kvm_regs, elr_el1);
+    vcpu.set_one_reg(arm64_core_reg_id!(KVM_REG_SIZE_U64, off), state.elr_el1)
+        .map_err(Error::SetCoreRegister)?;
+
+    let mut off = offset__of!(kvm_regs, spsr);
+    for i in 0..KVM_NR_SPSR as usize {
+        vcpu.set_one_reg(arm64_core_reg_id!(KVM_REG_SIZE_U64, off), state.spsr[i])
+            .map_err(Error::SetCoreRegister)?;
+        off += std::mem::size_of::<u64>();
+    }
+
+    let mut off = offset__of!(kvm_regs, fp_regs) + offset__of!(user_fpsimd_state, vregs);
+    for i in 0..32 {
+        vcpu.set_one_reg(
+            arm64_core_reg_id!(KVM_REG_SIZE_U128, off),
+            state.fp_regs.vregs[i][0],
+        )
+        .map_err(Error::SetCoreRegister)?;
+        off += mem::size_of::<u128>();
+    }
+
+    let off = offset__of!(kvm_regs, fp_regs) + offset__of!(user_fpsimd_state, fpsr);
+    vcpu.set_one_reg(
+        arm64_core_reg_id!(KVM_REG_SIZE_U32, off),
+        state.fp_regs.fpsr as u64,
+    )
+    .map_err(Error::SetCoreRegister)?;
+
+    let off = offset__of!(kvm_regs, fp_regs) + offset__of!(user_fpsimd_state, fpcr);
+    vcpu.set_one_reg(
+        arm64_core_reg_id!(KVM_REG_SIZE_U32, off),
+        state.fp_regs.fpcr as u64,
+    )
+    .map_err(Error::SetCoreRegister)?;
+    Ok(())
+}
+
+/// Get the state of the system registers.
+///
+/// # Arguments
+///
+/// * `vcpu` - Structure for the VCPU that holds the VCPU's fd.
+/// * `state` - Structure for returning the state of the system registers.
+pub fn save_system_registers(vcpu: &VcpuFd, state: &mut Vec<kvm_one_reg>) -> Result<()> {
+    // Call KVM_GET_REG_LIST to get all registers available to the guest. For ArmV8 there are
+    // around 500 registers.
+    let mut reg_list = RegList::new(512);
+    vcpu.get_reg_list(&mut reg_list)
+        .map_err(Error::GetRegList)?;
+
+    // At this point reg_list should contain: core registers and system registers.
+    // The register list contains the number of registers and their ids. We will be needing to
+    // call KVM_GET_ONE_REG on each id in order to save all of them. We carve out from the list
+    // the core registers which are represented in the kernel by kvm_regs structure and for which
+    // we can calculate the id based on the offset in the structure.
+
+    reg_list.retain(|regid| *regid != 0);
+    reg_list.as_slice().to_vec().sort();
+
+    reg_list.retain(|regid| is_system_register(*regid));
+
+    // Now, for the rest of the registers left in the previously fetched register list, we are
+    // simply calling KVM_GET_ONE_REG.
+    let indices = reg_list.as_slice();
+    for (_pos, index) in indices.iter().enumerate() {
+        if _pos > 230 {
+            break;
```

Comment:
Why ?

Reply:
This is a debug leftover also, removed it.